src/Surfnet/ServiceProviderDashboard/Domain/ValueObject/Ticket.php

```diff
@@ -37,14 +38,17 @@ class Ticket
     private $applicantName;
     /** @var string */
     private $applicantEmail;
+    /** @var string|null */
+    private $issueType;
 
-    public function __construct($entityId, $manageId, $nameEn, $applicantName, $applicantEmail)
+    public function __construct($entityId, $manageId, $nameEn, $applicantName, $applicantEmail, $issueType)
```

Comment:
I think it would be better if this parameter is mandatory to prevent a lot of if statements later on. Or perhaps remove it here if it's not the responsibility of the ticket.

Reply:
Still, I do not fully understand what you mean with it having to be mandatory. It is mandatory, right? No default value is set. And furthermore, an additional assertion is added to verify a non-empty string is set in the parameters.yml.

Please enlighten me :)

Reply:
It's suggested with [typehinting](https://github.com/SURFnet/sp-dashboard/blob/bbd1b4d9301a64dc59894ac1160e6bbf2c217440/src/Surfnet/ServiceProviderDashboard/Domain/ValueObject/Ticket.php#L41) and later on the issueType is [checked](https://github.com/SURFnet/sp-dashboard/blob/bbd1b4d9301a64dc59894ac1160e6bbf2c217440/src/Surfnet/ServiceProviderDashboard/Infrastructure/Jira/Factory/IssueFieldFactory.php#L117) if it's set.

[](url)

Reply:
Thanks, I've refactored some of the code. See the comment below for more information.